dali/kernels/signal/window/extract_windows_gpu.cuh

```diff
@@ -509,10 +509,11 @@ struct ExtractHorizontalWindowsImplGPU : ExtractWindowsImplGPU<Dst, Src> {
     constexpr int kDefaultBlockSize = 256;
     logical_block_size = kDefaultBlockSize;
     int64_t max_padded_length = 0;
-    int max_win_per_input = 0;
+    int max_win_per_input = 1;
```

Comment:
This seems wrong.

Reply:
In any case, 0 does not seem to be a valid value anyway: we divide by this value and use it in `clamp(val, 1, max_win_per_input)`. I thought that 0-sized batches cause the operator to be skipped completely, if that's not the case, this operator does not seem to be able to handle it right.

Reply:
I still think that keeping 0 makes sense - and we should rather `assert` or even just let it fail by dividing by 0 than proceed with a "valid" value that's not backed by the input data.
From security perspective, obviously wrong things (null dereference, division by 0) are safer than not-so-well defined situations.